src/main/java/de/rototor/pdfbox/graphics2d/PdfBoxGraphics2D.java

```diff
@@ -804,4 +845,14 @@ public FontRenderContext getFontRenderContext() {
 		calcGfx.addRenderingHints(renderingHints);
 		return calcGfx.getFontRenderContext();
 	}
+
+	// In case a single 2D graphic object has to be written (one or more times)
+	// into a PDF document, this method isn't needed.
+	// If, instead, several 2D graphic objects have to be written (one or more times)
+	// into a PDF document, the graphic state has to be saved after a graphic object
+	// has been written for the last time and a new graphic object has to be created
+	public void saveGraphicsState() throws IOException
```

Comment:
I'm aware that it may seem insane, but I do create a different PdfBoxGraphics2D object per chart (like in the example code I specified).
Actually I write each chart once, each one in a different page, with a different document content stream for each page!
I assume PDFBox (maybe it is a bug) cleans or releases some resources in the document when the graphic state is saved